packages/babel-core/src/config/config-chain.js

```diff
@@ -257,12 +272,14 @@ export function* buildRootChain(
   );
 
   return {
-    plugins: dedupDescriptors(chain.plugins),
-    presets: dedupDescriptors(chain.presets),
-    options: chain.options.map(o => normalizeOptions(o)),
+    plugins: isIgnored ? [] : dedupDescriptors(chain.plugins),
+    presets: isIgnored ? [] : dedupDescriptors(chain.presets),
+    options: isIgnored ? [] : chain.options.map(o => normalizeOptions(o)),
+    isIgnored,
```

Comment:
I would love if this was a string enum instead of a boolean.

In https://github.com/babel/babel/pull/11409 I was working on moving the extensions handling from `@babel/cli`/`@babel/register`/etc to `@babel/core` to support specifying file extensions in Babel config files.
However, I had a problem when making it work with the `--copy-files` and `--no-copy-ignored` options: my PR doesn't give a way of knowing _why_ a file is not transpiled (if it is ignored or if it has an unsupported extension).

If we made this option an enum, it would be then easy to extend it adding a third state.

Reply:
Something like `fileAction: 'transpile' | 'ignored' | 'unsupported'`? Not really sure what to name it haha. 😄

Reply:
I avoided suggesting any name on purpose, since I couldn't find one 😂

Maybe `fileHandling`, since "ignoring" isn't really an action but it's _not_ doing the action?

(Or maybe someone else will have a better alternative 🤞)

Reply:
I'm okay with `fileHandling`.

Reply:
Updated. Added `unsupported` to the type definition as well but it's not implemented yet. Can remove for now if you prefer.